field/TinyLang/Field/Uni.hs

```diff
@@ -73,20 +73,52 @@ type SomeUni f      = Some (Uni f)
 
 deriving instance Show (Uni f a)
 deriving instance Eq   (Uni f a)
+-- NOTE:  This instance fixes the uni (similarly to Eq)
+deriving instance Ord  (Uni f a)
 
-deriving instance Show (UniVar f a)
 deriving instance Show (SomeUniVar f)
 
--- This doesn't type check:
---
--- > UniConst _ x1 == UniConst _ x2 = x1 == x2
---
--- because it requires the type of @x1@ and @x2@ to have an @Eq@ instance.
--- We could provide a similar to 'withGeqUni' combinator that can handle this situation,
--- but then it's easier to just pattern match on universes.
-instance Eq f => Eq (UniVar f a) where
-    UniVar _ v1 == UniVar _ v2 = v1 == v2
-
+instance Eq (SomeUni f) where
+    Some u1 == Some u2 = withGeqUni u1 u2 False True
+
+instance Ord (SomeUni f) where
```

Comment:
You can probably derive [`GCompare`](http://hackage.haskell.org/package/dependent-sum-0.5/docs/Data-GADT-Compare.html#t:GCompare) and define `Ord` in terms of it.